### ISSUE TITLE
Server responds with HTTP 415 for GET requests which include Content-Type header

### ISSUE THREAD

**Issue by mcmasn-amzn** [awslabs/smithy-rs#1845]
After upgrading from 0.48 to 0.49, I started getting HTTP 415 for GET calls that include the Content-Type header. My interpretation of the RFCs on this header https://httpwg.org/specs/rfc9110.html#representation.metadata is that this behavior falls into an unspecified area.

It makes sense that -- strictly speaking -- this header should not be attached to requests without a message body.

But on the other hand, a common approach is to make servers lenient and ignore headers that don't matter. So, would it be better if the SmithyRs server simply ignored the "Content-Type" header for GET requests instead?

**Comment by 82marbag** [comment]
Does this happen only when your model input is empty? It's forbidden [here](https://github.com/awslabs/smithy/blob/603c0f2de0dc54c7ed5cfa0d1e2f1c63d521eed2/smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy#L14). This could be a question for the smithy team

**Comment by mcmasn-amzn** [comment]
Yes. That's an odd quirk. I noticed it happening on a "ping" API that I implemented with an empty operation

```
@http(uri: "/health-check", method: "GET")
operation GetHealthCheck {}
```
```
curl -i http://localhost:3000/health-check -H 'Content-Type: application/json'
HTTP/1.1 415 Unsupported Media Type
content-type: application/json
x-amzn-errortype: UnsupportedMediaTypeException
content-length: 2
date: Fri, 14 Oct 2022 05:57:57 GMT

{}
```

But the issue goes away if I add a dummy input structure.

```
@http(uri: "/health-check", method: "GET")
operation HealthCheckOperation {
    input:={
        @httpQuery("param1")
        param1: String
    }
}
```
```
curl -i http://localhost:3000/health-check -H 'Content-Type: application/json'
HTTP/1.1 200 OK
content-type: application/json
content-length: 0
date: Fri, 14 Oct 2022 06:07:50 GMT
```

**Comment by 82marbag** [comment]
We're complying with the Smithy spec here, it's not a smithy-rs specific behavior. You could ask to Smithy: https://github.com/awslabs/smithy

**Comment by 82marbag** [comment]
Note that if your operation has an empty input struct, this behavior does not apply. It does apply only when there is no input in the model

**Comment by mcmasn-amzn** [comment]
Yes, it would be good to clarify with Smithy because the spec you linked to [here](https://github.com/awslabs/smithy/blob/603c0f2de0dc54c7ed5cfa0d1e2f1c63d521eed2/smithy-aws-protocol-tests/model/restJson1/malformedRequests/malformed-content-type.smithy#L14-L17) only shows examples of POST requests.

In my case, I'm dealing with GET requests. It seems like inconsistent behavior from a user perspective. The exact same `GET http://localhost:3000/health-check` request might pass or fail, but it depends on whether there is something defined on the input model.

**Comment by mcmasn-amzn** [comment]
https://github.com/awslabs/smithy/issues/1450
